server/src/main/java/org/elasticsearch/health/metadata/HealthMetadataService.java

```diff
@@ -53,25 +54,29 @@ public class HealthMetadataService {
     private volatile boolean enabled;
 
     // Signifies that a node has been elected as master, but it was not able yet to publish its health metadata for
-    // other reasons for example not all nodes of the cluster are 8.4.0 or newer
-    private volatile boolean readyToPublish = false;
+    // other reasons for example not all nodes of the cluster are 8.5.0 or newer
+    private volatile boolean readyToPublishAsMaster = false;
     // Allows us to know if this node is the elected master without checking the cluster state, effectively protecting
     // us from checking the cluster state before the cluster state is initialized
     private volatile boolean isMaster = false;
+    private volatile HealthMetadata previousHealthMetadata;
```

Comment:
I am wondering if we should keep the `previousHealthMetadata` in memory or we should better retrieve it from the cluster state each time. I am concerned there might be rare scenarios that we an update is lost and the "previous" would diverge from what is currently on cluster state which I believe is the "true" previous... What do you think?